common/tools/dev-tool/src/util/samples/processor.ts

```diff
@@ -0,0 +1,443 @@
+      statement.modifiers.some(({ kind }) => kind === ts.SyntaxKind.ExportKeyword);
+
+    if (!isDefault) {
+      return statement;
+    }
+
+    // The only forms that can have `export default` modifiers are the following.
+    const decl = statement as ts.FunctionDeclaration | ts.ClassDeclaration;
+    const updatedModifiers = decl.modifiers?.filter(
+      ({ kind }) => kind !== ts.SyntaxKind.DefaultKeyword && kind !== ts.SyntaxKind.ExportKeyword
+    );
+
+    if (!decl.name) {
+      // If there is no name, the declaration is anonymous, and we will bind it as an expression in module.exports
+      const initializer = ts.isClassDeclaration(decl)
+        ? factory.createClassExpression(
+            decl.decorators,
+            updatedModifiers,
+            undefined,
+            decl.typeParameters,
+            decl.heritageClauses,
+            decl.members
+          )
+        : decl.body === undefined // This is a strange case that I assume has to do with overload declarations.
+        ? undefined
+        : factory.createFunctionExpression(
+            updatedModifiers,
+            decl.asteriskToken,
+            undefined,
+            decl.typeParameters,
+            decl.parameters,
+            decl.type,
+            decl.body
+          );
+
+      if (initializer) {
+        exportEntries.push(factory.createPropertyAssignment("default", initializer));
+      }
+
+      return factory.createEmptyStatement();
+    }
+
+    exportEntries.push(factory.createPropertyAssignment("default", decl.name));
+
+    return ts.isClassDeclaration(decl)
+      ? factory.updateClassDeclaration(
+          decl,
+          decl.decorators,
+          updatedModifiers,
+          decl.name,
+          decl.typeParameters,
+          decl.heritageClauses,
+          decl.members
+        )
+      : factory.updateFunctionDeclaration(
+          decl,
+          decl.decorators,
+          updatedModifiers,
+          decl.asteriskToken,
+          decl.name,
+          decl.typeParameters,
+          decl.parameters,
+          decl.type,
+          decl.body
+        );
+  });
+}
+
+/**
+ * Determines whether a module specifier is a package dependency.
+ *
+ * A dependency is a module specifier that does not refer to a node builtin and
+ * is not a relative path.
+ *
+ * Absolute path imports are not supported in samples (because the package base
+ * is not fixed relative to the source file).
+ *
+ * @param moduleSpecifier - the string given to `import` or `require`
+ * @returns - true if `moduleSpecifier` should be considered a reference to a
+ * node module dependency
+ */
+function isDependency(moduleSpecifier: string): boolean {
+  if (nodeBuiltins.includes(moduleSpecifier)) return false;
+
+  return !isRelativePath(moduleSpecifier);
+}
+
+// This seems like a reasonable test for "is a relative path" as long as
+// absolute path imports are forbidden.
+const isRelativePath = (path: string) => /^\.\.?[\/\\]/.test(path);
```

Comment:
Would this work here? https://nodejs.org/api/path.html#pathisabsolutepath

Reply:
An absolute path in an import statement isn't forbidden, but will break the samples in some other way. I'm really only concerned with relative paths in the sample folder.